### PR TITLE
Feat/lib 446 dialog component

### PR DESCRIPTION
**Pull request by Trebh**
[Jira](https://fiscozen.atlassian.net/browse/LIB-446)

### CODE REVIEW

File: apps/storybook/tailwind.config.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ export default {
     ...tailwindConfig,
     content: [
         "./src/**/*.{html,js,ts,vue}", 
-        "./node_modules/@fiscozen/**/src/**/*.{html,js,ts,vue}"
+        "./node_modules/@fiscozen/**/src/**/*.{html,js,ts,vue}",
+        "./node_modules/@fiscozen/dialog/src/**/*.{html,js,ts,vue}",
```
Comment (Crish15) [inline apps/storybook/tailwind.config.js:9]: with the new version of Storybook this inclusion should not be necessary

File: packages/dialog/src/FzSimpleDialog.vue
Comment (oxcened) [inline packages/dialog/src/FzSimpleDialog.vue:1]: FzConfirmDialog could be a more descriptive name maybe

File: packages/dialog/src/types.ts [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  /**
+   * Text content in the cancel button
+   */
+  cancelLabel?: string;
+  /**
+   * Text content in the body slot
+   */
+  text?: string;
+  /**
+   * Wether to show the modal as a drawer
```
Comment (oxcened) [inline packages/dialog/src/types.ts:25]:
```suggestion
   * Whether to show the modal as a drawer
```

File: packages/dialog/src/FzDialog.vue [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+        "h-screen md:h-auto": true,
+      };
+      break;
+    case "xl":
+      res = {
+        "w-screen xl:w-[960px]": true,
+        "min-h-[400px]": true,
+        "xl:max-h-[600px]": true,
+        "h-screen xl:h-auto": true,
+      };
```
Comment (oxcened) [inline packages/dialog/src/FzDialog.vue:160]: an object is redundant as everything has a true value, you could simplify it by using an array

File: packages/dialog/src/FzDialog.vue [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+        "xs:max-md:w-screen": true,
+        "xs:max-md:max-w-screen-xl": true,
+      };
+    case "xl":
+      res = {
+        "xs:max-xl:m-0": true,
+        "xs:max-xl:max-h-screen": true,
+        "xs:max-xl:h-screen": true,
+        "xs:max-xl:w-screen": true,
+        "xs:max-xl:max-w-screen-xl": true,
```
Comment (oxcened) [inline packages/dialog/src/FzDialog.vue:111]: same here, array is a more appropriate structure

File: packages/dialog/src/FzDialog.vue [hunk 5]
```diff
@@ -0,0 +1,174 @@
...
+      };
+      break;
+      break;
```
Comment (oxcened) [inline packages/dialog/src/FzDialog.vue:114]: two breaks

File: packages/dialog/src/FzDialog.vue [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+const props = withDefaults(defineProps<FzDialogProps>(), {
+  size: "md",
+});
+const emit = defineEmits(["confirm", "cancel"]);
+
+const dialog = ref<HTMLDialogElement>();
+const visible = ref(false);
+
+const showModal = () => {
+  dialog.value?.showModal();
```
Comment (oxcened) [inline packages/dialog/src/FzDialog.vue:30]: I think we should expect dialog.value to exist

File: packages/dialog/src/FzDialog.vue [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+defineExpose({
+  show: showModal,
+  close: (returnVal?: string): void => dialog.value?.close(returnVal),
+  visible,
+});
+
+const handleBackdropClick = (event: MouseEvent) => {
+  if (!dialog.value) {
+    return;
```
Comment (oxcened) [inline packages/dialog/src/FzDialog.vue:42]: same here

File: packages/dialog/src/FzSimpleDialog.vue [hunk 8]
```diff
@@ -0,0 +1,83 @@
...
+  "h-32": !props.isDrawer,
+  "h-40": props.isDrawer,
+};
+
+const show = () => {
+  dialog.value?.show();
+  visible.value = true;
+};
+
+const cancel = () => {
```
Comment (oxcened) [inline packages/dialog/src/FzSimpleDialog.vue:65]:
```suggestion
const handleCancel = () => {
```

File: packages/dialog/src/FzSimpleDialog.vue [hunk 9]
```diff
@@ -0,0 +1,83 @@
...
+  dialog.value?.close();
+  visible.value = false;
+  emit("fzmodal:cancel");
+};
+
+const confirm = () => {
```
Comment (oxcened) [inline packages/dialog/src/FzSimpleDialog.vue:71]:
```suggestion
const handleConfirm = () => {
```

File: packages/dialog/src/FzSimpleDialog.vue [hunk 10]
```diff
@@ -0,0 +1,83 @@
...
+  "h-40": props.isDrawer,
+};
+
+const show = () => {
+  dialog.value?.show();
+  visible.value = true;
+};
+
+const cancel = () => {
+  dialog.value?.close();
```
Comment (oxcened) [inline packages/dialog/src/FzSimpleDialog.vue:66]: same here

File: packages/dialog/src/FzSimpleDialog.vue [hunk 11]
```diff
@@ -0,0 +1,83 @@
...
+  visible.value = false;
+  emit("fzmodal:cancel");
+};
+
+const confirm = () => {
+  dialog.value?.close();
```
Comment (oxcened) [inline packages/dialog/src/FzSimpleDialog.vue:72]: same here

File: packages/dialog/src/FzDialog.vue (hunk 3 above)
Comment (Trebh) [reply packages/dialog/src/FzDialog.vue:160]: fair enough

**Review by oxcened — APPROVED** [review]

**Pull request merged**